contracts/core/RebalancingSet.sol

```diff
@@ -0,0 +1,389 @@
+     * Constructor function for Rebalancing Set Token
+     *
+     *
+     * @param _factory                      The factory used to create the Rebalancing Set
+     * @param _manager                      The manager of the Rebalancing Set
+     * @param _initialSet                   The initial set that collateralizes the Rebalancing set
+     * @param _initialShareRatio            How much of a Set (in gWei) equals one share
+     * @param _proposalPeriod               Amount of time for users to inspect a rebalance proposal
+     * @param _rebalanceInterval            The minimum amount of time between rebalances
+     * @param _name                         The Rebalancing Set's name
+     * @param _symbol                       The Rebalancing Set's symbol
+     */
+
+    constructor(
+        address _factory,
+        address _manager,
+        address _initialSet,
+        uint256 _initialShareRatio,
+        uint256 _proposalPeriod,
+        uint256 _rebalanceInterval,
+        string _name,
+        string _symbol
+    )
+        public
+        DetailedERC20(_name, _symbol, 18)
+    {
+        // Require day long proposal period
+        require(_proposalPeriod > 86400);
+
+        // Require one day between end of rebalance and proposing another rebalance
+        require(_rebalanceInterval > 86400);
+
+        factory = _factory;
+        manager = _manager;
+        currentSet = _initialSet;
+        shareRatio = _initialShareRatio;
+
+        proposalPeriod = _proposalPeriod;
+        rebalanceInterval = _rebalanceInterval;
+        lastRebalanceTimestamp = block.timestamp;
+        rebalanceState = State.Default;
+    }
+
+    /* ============ Public Functions ============ */
+
+    /**
+     * Function used to set the terms of the next rebalance and start the proposal period
+     *
+     *
+     * @param _rebalancingSet               The Set to rebalance into
+     * @param _auctionLibrary               The library used to calculate the Dutch Auction price
+     * @param _curveCoefficient             The slope (or convexity) of the price curve
+     * @param _auctionPriceDivisor          The granularity with which the prices change
+     * @param _auctionStartPrice            The price to start the auction at
+     */
+    function propose(
+        address _rebalancingSet,
+        address _auctionLibrary,
+        uint256 _curveCoefficient,
+        uint256 _auctionStartPrice,
+        uint256 _auctionPriceDivisor
+    )
+        external
+    {
+        // Make sure it is manager that is proposing the rebalance
+        require(msg.sender == manager);
+
+        // New proposal cannot be made during a rebalance period
+        require(rebalanceState == State.Default);
+
+        // Make sure enough time has passed from last rebalance to start a new proposal
+        require(block.timestamp >= lastRebalanceTimestamp.add(rebalanceInterval));
+
+        // Set auction parameters
+        rebalancingSet = _rebalancingSet;
```

Comment:
We should have a check to ensure that the address proposed is indeed a Set, perhaps querying Core

Reply:
Yeah I'd be ok with that.